presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -336,6 +341,22 @@ public Expression rewriteLambdaExpression(LambdaExpression node, Void context, E
             {
                 return new LambdaExpression(node.getArguments(), treeRewriter.rewrite(node.getBody(), context));
             }
+
+            @Override
+            public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+            {
+                if (node.getName().equals(QualifiedName.of("$INTERNAL$DEFERRED_SYMBOL_REFERENCE"))) {
```

Comment:
What is `$INTERNAL$DEFERRED_SYMBOL_REFERENCE"` and why it is needed? I though we got rid of them?

I understand there is no SQL syntax for `DynamicFilterExpression` but it would be serialized to SQL:
```
$INTERNAL$dynamicFilter(...)
```
and then deserialized to `DynamicFilterExpression`

Reply:
> This part was needed when DynamicFilter was represented as a function call

I still don't see why this code is needed here. There is nothing special with `dynamicFilter` function call, it just contains `expression, varchar, varchar`.

Alternatively, for `DynamicFilter` dedicated expression this also shouldn't be needed. `DynamicFilter` could be serialized to `$INTERNAL$DYNAMIC_FILTER` again with standard SQL parameters. This should be handled by expression serialization

Reply:
Dereference symbol was left out accidentally when rebasing stuff.